tests/components/application-connector/internal/testkit/test-api/apis.go

```diff
@@ -49,29 +49,49 @@ func SetupRoutes(logOut io.Writer, basicAuthCredentials BasicAuthCredentials, oA
 	{
 		r := api.PathPrefix("/csrf-basic").Subrouter()
 		r.Use(csrf.Middleware())
-		r.Use(BasicAuth(basicAuthCredentials.User, basicAuthCredentials.Password))
+		r.Use(BasicAuth(basicAuthCredentials))
 		r.HandleFunc("/ok", alwaysOk).Methods(http.MethodGet)
 		r.HandleFunc("/echo", echo)
 	}
 	{
 		r := api.PathPrefix("/request-parameters-basic").Subrouter()
-		r.Use(RequestParameters(expectedRequestParameters.Headers, expectedRequestParameters.QueryParameters))
-		r.Use(BasicAuth(basicAuthCredentials.User, basicAuthCredentials.Password))
+		r.Use(RequestParameters(expectedRequestParameters))
+		r.Use(BasicAuth(basicAuthCredentials))
 		r.HandleFunc("/ok", alwaysOk).Methods(http.MethodGet)
 		r.HandleFunc("/echo", echo)
 	}
 
 	return router
 }
 
-type BasicAuthCredentials struct {
-	User     string
-	Password string
+func SetupMTLSRoutes(logOut io.Writer, oAuthCredentials OAuthCredentials, tokens map[string]OAuthToken) http.Handler {
+	router := mux.NewRouter()
+
+	router.HandleFunc("/v1/health", alwaysOk).Methods("GET")
+	api := router.PathPrefix("/v1/api").Subrouter()
+	api.Use(Logger(logOut, logger.DevLoggerType))
+
+	oauth := NewOAuth(oAuthCredentials.ClientID, oAuthCredentials.ClientSecret, tokens)
+
+	{
+		r := api.PathPrefix("/mtls").Subrouter()
+		r.Use(oauth.Middleware())
+		api.HandleFunc("/mtls-oauth/token", oauth.MTLSToken).Methods(http.MethodPost)
```

Comment:
What do you think about adding the `/ok` and `/echo` endpoints here just o be consistent with the other routers? The test could use `targetUrl: "http://{{ .Values.mockServiceName }}.{{ .Values.namespace }}.svc.cluster.local:8080/v1/api/mtls-oauth/ok"` instead of just `/auth/ok` where it comes to the MTLS tests. In this case, I believe, there is no need to pass shared `tokens` to the `SetupMTLSRoutes` and `SetupRoutes`, and initialize `NewOAuth` with them
```go
r.HandleFunc("/ok", alwaysOk).Methods(http.MethodGet)
r.HandleFunc("/echo", echo)
```

Reply:
Discussed, my bad :D